constexpr_init.pass.cpp (tests/libcxx) fails to compile with GCC 7

GCC complains:
```
/home/ubuntu/openenclave/tests/libcxx/enc/../../../3rdparty/libcxx/libcxx/test/std/utilities/function.objects/comparisons/constexpr_init.pass.cpp:42:1: error: non-constant condition for static assertion
 static_assert(test_constexpr_context<int>(), "");
 ^~~~~~~~~~~~~
/home/ubuntu/openenclave/tests/libcxx/enc/../../../3rdparty/libcxx/libcxx/test/std/utilities/function.objects/comparisons/constexpr_init.pass.cpp:42:42: error: ‘constexpr bool test_constexpr_context() [with T = int]’ called in a constant expression
 static_assert(test_constexpr_context<int>(), "");
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/ubuntu/openenclave/tests/libcxx/enc/../../../3rdparty/libcxx/libcxx/test/std/utilities/function.objects/comparisons/constexpr_init.pass.cpp:26:16: note: ‘constexpr bool test_constexpr_context() [with T = int]’ is not usable as a constexpr function because:
 constexpr bool test_constexpr_context() {
                ^~~~~~~~~~~~~~~~~~~~~~
/home/ubuntu/openenclave/tests/libcxx/enc/../../../3rdparty/libcxx/libcxx/test/std/utilities/function.objects/comparisons/constexpr_init.pass.cpp:27:20: error: uninitialized variable ‘eq’ in ‘constexp ’ function
   std::equal_to<T> eq;
                    ^~
/home/ubuntu/openenclave/tests/libcxx/enc/../../../3rdparty/libcxx/libcxx/test/std/utilities/function.objects/comparisons/constexpr_init.pass.cpp:43:1: error: non-constant condition for static assertion
 static_assert(test_constexpr_context<void>(), "");
 ^~~~~~~~~~~~~
/home/ubuntu/openenclave/tests/libcxx/enc/../../../3rdparty/libcxx/libcxx/test/std/utilities/function.objects/comparisons/constexpr_init.pass.cpp:43:43: error: ‘constexpr bool test_constexpr_context() [with T = void]’ called in a constant expression
 static_assert(test_constexpr_context<void>(), "");
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/ubuntu/openenclave/tests/libcxx/enc/../../../3rdparty/libcxx/libcxx/test/std/utilities/function.objects/comparisons/constexpr_init.pass.cpp:26:16: note: ‘constexpr bool test_constexpr_context() [with T = void]’ is not usable as a constexpr function because:
 constexpr bool test_constexpr_context() {
                ^~~~~~~~~~~~~~~~~~~~~~
/home/ubuntu/openenclave/tests/libcxx/enc/../../../3rdparty/libcxx/libcxx/test/std/utilities/function.objects/comparisons/constexpr_init.pass.cpp:27:20: error: uninitialized variable ‘eq’ in ‘constexp ’ function
   std::equal_to<T> eq;
```

This is due to:
https://github.com/Microsoft/openenclave/blob/master/3rdparty/libcxx/libcxx/test/std/utilities/function.objects/comparisons/constexpr_init.pass.cpp#L19

Target milestone for this issue to be resolved is GCC 7.5.0 as per: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=83921

--- a/tests/libcxx/CMakeLists.txt
+++ b/tests/libcxx/CMakeLists.txt
@@ -59,11 +59,11 @@ foreach(testcase ${alltests})
         endif()
     endif()
 
-# the following test fails on GCC version 7, see  #1523 -- Skipping this test in GCC 7.0 -> 7.5
+# the following test fails on GCC version 7, see  #1523 -- Skipping this test in GCC 7.0 -> 8.0
     if ("${name}" MATCHES "std_utilities_function.objects_comparisons_constexpr_init.pass")
         if(ENCLAVE_CXX_COMPILER_ID MATCHES GNU
             AND ENCLAVE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7
-            AND ENCLAVE_CXX_COMPILER_VERSION VERSION_LESS 7.5)
+            AND ENCLAVE_CXX_COMPILER_VERSION VERSION_LESS 8.0)
             continue()
         endif()
     endif()

--- a/tests/libcxx/enc/CMakeLists.txt
+++ b/tests/libcxx/enc/CMakeLists.txt
@@ -144,11 +144,11 @@ foreach(testcase ${alltests})
         endif()
     endif()
 
-# the following test fails on GCC version 7, see  #1523 -- Skipping this test in GCC 7.0 -> 7.5
+# the following test fails on GCC version 7, see  #1523 -- Skipping this test in GCC 7.0 -> 8.0
     if ("${name}" MATCHES "std_utilities_function.objects_comparisons_constexpr_init.pass")
         if(CMAKE_CXX_COMPILER_ID MATCHES GNU
                AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7
-               AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.5)
+               AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.0)
             continue()
         endif()
     endif()

